processor/metricstransformprocessor/metrics_transform_processor.go

```diff
@@ -65,7 +66,26 @@ func (mtp *metricsTransformProcessor) ProcessMetrics(_ context.Context, md pdata
 			nameToMetricMapping[metric.MetricDescriptor.Name] = metric
 		}
 
+		resourceAttributes := make(map[string]string)
+		if data.Resource != nil {
+			resourceAttributes = data.Resource.Labels
+		}
 		for _, transform := range mtp.transforms {
+			transform.resourceAttributeMap = resourceAttributes
+
+			// When user sets "metric_name=all_metrics", these operations are going to be applied
+			// to all metric data points. Right now, this batch operation only works and is tested for
+			// `convert_resource_attributes_to_labels` operation, and does not break existing experience
+			// for other operations. After updating the metrics, it will continue for the next iteration.
+			// TODO: add more unit tests and update README to pubicly anounce this feature for other
+			// operations as well.
+			if transform.MetricName == "all_metrics" {
```

Comment:
Rather than use "all_metrics" as a special value, I think it would make more sense to do what @tigrannajaryan suggested and make `MetricName` optional in the config. If left out, then the transform applies to all metrics.

Reply:
Pushed an update. Nice suggestion. Thanks @james-bebbington.